platforms/defaults_windows.go

```diff
@@ -92,3 +101,28 @@ func prefix(v string) string {
 func Default() MatchComparer {
 	return Only(DefaultSpec())
 }
+
+var (
+	ubr  int
+	once sync.Once
+)
+
+// there is no windows system call in golang yet to get host UBR, so we do our own.
+func getHostWindowsUpdateBuildRevision() int {
+	once.Do(func() {
+		ubr = 0
```

Comment:
Possibly should be -1? 0 is a possible UBR, sometimes seen in internal builds, e.g., https://betawiki.net/wiki/Windows_11_build_22451. I don't think we will see this value anywhere where it wouldn't be clear something has failed if you saw -1.